salt/states/glusterfs.py

```diff
@@ -48,57 +49,59 @@ def peered(name):
            'comment': '',
            'result': False}
 
+    # Check if the name resolves to localhost
+    if socket.gethostbyname(name) in __salt__['network.ip_addrs']():
+        ret['result'] = True
+        ret['comment'] = 'Peering with localhost is not needed'
+        return ret
+
     peers = __salt__['glusterfs.list_peers']()
 
-    if peers:
-        if name in peers:
-            ret['result'] = True
-            ret['comment'] = 'Host {0} already peered'.format(name)
-            return ret
-        elif __opts__['test']:
-            ret['comment'] = 'Peer {0} will be added.'.format(name)
-            ret['result'] = None
-            return ret
+    if peers and name in peers:
+        ret['result'] = True
+        ret['comment'] = 'Host {0} already peered'.format(name)
+        return ret
 
     if suc.check_name(name, 'a-zA-Z0-9._-'):
         ret['comment'] = 'Invalid characters in peer name.'
         ret['result'] = False
         return ret
 
-    if 'output' in __salt__['glusterfs.peer'](name):
-        ret['comment'] = __salt__['glusterfs.peer'](name)['output']
-    else:
-        ret['comment'] = ''
+    if __opts__['test']:
+        ret['comment'] = 'Peer {0} will be added.'.format(name)
+        ret['result'] = None
+        return ret
 
+    peered = __salt__['glusterfs.peer'](name)
+    if not peered:
+        ret['comment'] = 'Failed to peer with {0}, please check logs for errors'.format(name)
+        return ret
+
+    # Double check that the action succeeded
     newpeers = __salt__['glusterfs.list_peers']()
-    #if newpeers was null, we know something didn't work.
     if newpeers and name in newpeers:
         ret['result'] = True
+        ret['comment'] = 'Host {0} successfully peered'.format(name)
         ret['changes'] = {'new': newpeers, 'old': peers}
-    #In case the hostname doesn't have any periods in it
-    elif name == socket.gethostname():
-        ret['result'] = True
-        return ret
-    #In case they have a hostname like "example.com"
-    elif name == socket.gethostname().split('.')[0]:
-        ret['result'] = True
-        return ret
-    elif 'on localhost not needed' in ret['comment']:
-        ret['result'] = True
-        ret['comment'] = 'Peering with localhost is not needed'
     else:
-        ret['result'] = False
+        ret['comment'] = 'Host {0} was successfully peered but did not appear in the list of peers'.format(name)
     return ret
 
 
-def created(name, bricks, stripe=False, replica=False, device_vg=False,
+def volume_present(name, bricks, stripe=False, replica=False, device_vg=False,
```

Comment:
I'm in the process of completely reworking this to be much more intelligent, allowing you to change the volume definition by just passing it a changed state tree, eg add/remove/replace bricks/servers/replica levels, etc.